### ISSUE TITLE
Host-dns breaks hubble-relay

### ISSUE THREAD

**Issue by soulwhisper** [siderolabs/talos#8661]
### Description
set up hostDNS with `forwardKubeDNSToHost: true` and `resolveMemberNames: true` breaks cilium hubble-relay.

### Logs
After cilium helm deployment with parameters 
```
helm install cilium cilium/cilium \
    --version 1.15.4 \
    --namespace kube-system \
    --set ipam.mode=kubernetes \
    --set kubeProxyReplacement=true \
    --set securityContext.capabilities.ciliumAgent="{CHOWN,KILL,NET_ADMIN,NET_RAW,IPC_LOCK,SYS_ADMIN,SYS_RESOURCE,DAC_OVERRIDE,FOWNER,SETGID,SETUID}" \
    --set securityContext.capabilities.cleanCiliumState="{NET_ADMIN,SYS_ADMIN,SYS_RESOURCE}" \
    --set cgroup.autoMount.enabled=false \
    --set cgroup.hostRoot=/sys/fs/cgroup \
    --set k8sServiceHost=localhost \
    --set k8sServicePort=7445 \
    --set prometheus.enabled=true \
    --set operator.prometheus.enabled=true \
    --set hubble.enabled=true \
    --set hubble.relay.enabled=true \
    --set hubble.ui.enabled=true \
    --set hubble.metrics.enableOpenMetrics=true \
    --set hubble.metrics.enabled="{dns,drop,tcp,flow,port-distribution,icmp,httpV2:exemplars=true;labelsContext=source_ip\,source_namespace\,source_workload\,destination_ip\,destination_namespace\,destination_workload\,traffic_direction}"
```
hubble-relay will break with message
```
Message:   time="2024-04-26T01:48:42Z" level=info msg="Starting gRPC health server..." addr=":4222" subsys=hubble-relay
time="2024-04-26T01:48:42Z" level=info msg="Starting gRPC server..." options="{peerTarget:hubble-peer.kube-system.svc.cluster.local:443 dialTimeout:5000000000 retryTimeout:30000000000 listenAddress::4245 healthListenAddress::4222 metricsListenAddress: log:0xc0002e9a40 serverTLSConfig:<nil> insecureServer:true clientTLSConfig:0xc0006ed920 clusterName:default insecureClient:false observerOptions:[0x1f0ed60 0x1f0ee40] grpcMetrics:<nil> grpcUnaryInterceptors:[] grpcStreamInterceptors:[]}" subsys=hubble-relay
time="2024-04-26T01:48:47Z" level=warning msg="Failed to create peer client for peers synchronization; will try again after the timeout has expired" error="context deadline exceeded" subsys=hubble-relay target="hubble-peer.kube-system.svc.cluster.local:443"
time="2024-04-26T01:49:22Z" level=warning msg="Failed to create peer client for peers synchronization; will try again after the timeout has expired" error="context deadline exceeded" subsys=hubble-relay target="hubble-peer.kube-system.svc.cluster.local:443"
time="2024-04-26T01:49:42Z" level=info msg="Stopping server..." subsys=hubble-relay
time="2024-04-26T01:49:42Z" level=info msg="Server stopped" subsys=hubble-relay
```
Similar issue and debug process described [here](https://github.com/cilium/cilium/issues/31516) and [here](https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/)
Debugging screenshots below:
![image](https://github.com/siderolabs/talos/assets/26273512/65ce0217-b02e-4bf8-a2f3-eddab1b8e7e8)

### Environment

- Talos version: 1.7
- Kubernetes version: 1.30.0
- Platform: Vsphere 7.0

**Comment by soulwhisper** [comment]
Seems related to commit [2bf613a](https://github.com/siderolabs/talos/commit/2bf613ad3bd1582b520b2f661b7e0bfab4207eed)

**Comment by smira** [comment]
The fix will be backported to Talos v1.7.1, let's see if that fixes it.

The option `forwardKubeDNSToHost` is not enabled by default yet.
